### DISCUSSION TITLE
Calling API request from component vs. API route

### DISCUSSION THREAD

**Discussion by johnnypea** [vercel/next.js#17567] [Help]
What are the benefits calling API request from API route as shown here? https://github.com/vercel/swr/tree/master/examples/basic-typescript

What are the caveats?

Thanks.

**Comment by leerob** [comment]
I think that's purely for demonstration, but I can provide one great example for API routes: environment variables. 

API routes allow you to easily communicate with external APIs in a secure fashion. You wouldn't want to expose those environment variables on the client-side, so API routes allow you to "wrap" calling an external API and format the data to your liking. I've used this quite a bit.
